### PR TITLE
[MAINTENANCE] Update error message around `context.add_checkpoint` when neither name nor checkpoint are provided

### PR DESCRIPTION
**Pull request by cdkini**
Update to messaging depending on what args are passed in

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8313&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=8313&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -963,9 +963,10 @@ def _validate_add_datasource_args(
         datasource: BaseDatasource | FluentDatasource | LegacyDatasource | None,
     ) -> None:
         if not ((datasource is None) ^ (name is None)):
-            raise ValueError(
-                "Must either pass in an existing datasource or individual constructor arguments (but not both)"
-            )
+            error_message = "Must either pass in an existing datasource or individual constructor arguments"
+            if datasource and name:
+                error_message += " (but not both)"
+            raise ValueError(error_message)
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py]: I think this should be a `TypeError`
If you fail to provide a required positional argument you'll get a `TypeError`
```python
def foo(arg1, arg2):
    pass

foo()

... TypeError: foo() missing 2 required positional arguments: 'arg1' and 'arg2'
```

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 2]
```diff
@@ -963,9 +963,10 @@ def _validate_add_datasource_args(
         datasource: BaseDatasource | FluentDatasource | LegacyDatasource | None,
     ) -> None:
         if not ((datasource is None) ^ (name is None)):
-            raise ValueError(
-                "Must either pass in an existing datasource or individual constructor arguments (but not both)"
-            )
+            error_message = "Must either pass in an existing datasource or individual constructor arguments"
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py]:
```suggestion
            error_message = "Must either pass in an existing 'datasource' or individual constructor arguments"
```

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 3]
```diff
@@ -2255,9 +2256,10 @@ def _resolve_add_checkpoint_args(  # noqa: PLR0913
         from great_expectations.checkpoint.checkpoint import Checkpoint
 
         if not ((checkpoint is None) ^ (name is None)):
-            raise ValueError(
-                "Must either pass in an existing checkpoint or individual constructor arguments (but not both)"
-            )
+            error_message = "Must either pass in an existing checkpoint or individual constructor arguments"
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py]:
```suggestion
            error_message = "Must either pass in an existing 'checkpoint' or individual constructor arguments"
```

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 4]
```diff
@@ -963,9 +963,10 @@ def _validate_add_datasource_args(
         datasource: BaseDatasource | FluentDatasource | LegacyDatasource | None,
     ) -> None:
         if not ((datasource is None) ^ (name is None)):
-            raise ValueError(
-                "Must either pass in an existing datasource or individual constructor arguments (but not both)"
-            )
+            error_message = "Must either pass in an existing 'datasource' or individual constructor arguments"
+            if datasource and name:
+                error_message += " (but not both)"
+            raise ValueError(error_message)
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py]:
```suggestion
            raise TypeError(error_message)
```

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 5]
```diff
@@ -2255,9 +2256,10 @@ def _resolve_add_checkpoint_args(  # noqa: PLR0913
         from great_expectations.checkpoint.checkpoint import Checkpoint
 
         if not ((checkpoint is None) ^ (name is None)):
-            raise ValueError(
-                "Must either pass in an existing checkpoint or individual constructor arguments (but not both)"
-            )
+            error_message = "Must either pass in an existing 'checkpoint' or individual constructor arguments"
+            if checkpoint and name:
+                error_message += " (but not both)"
+            raise ValueError(error_message)
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py]:
```suggestion
            raise TypeError(error_message)
```

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**